app/scripts/views/base.js

```diff
@@ -668,6 +668,40 @@ function (Cocktail, _, Backbone, Raven, $, p, AuthErrors,
      */
     showSubView: function (/* SubView */) {
       // Implement in subclasses
+    },
+
+    /**
+     * Invoke a method on the broker, handling any returned behaviors
+     *
+     * @method invokeBrokerMethod
+     * @param {string} methodName
+     * @param ...
+     * @return {promise}
+     */
+    invokeBrokerMethod: function (methodName/*, ...*/) {
+      var args = [].slice.call(arguments, 1);
+
+      var self = this;
+      var broker = self.broker;
+
+      return p(broker[methodName].apply(broker, args))
+        .then(self.invokeBehavior.bind(self));
+    },
+
+    /**
+     * Invoke a behavior returned by an auth broker.
+     *
+     * @method invokeBehavior
+     * @param {function} behavior
+     * @return {variant} behavior's return value if behavior is a function,
+     *         otherwise return the behavior.
+     */
+    invokeBehavior: function (behavior) {
+      if (typeof behavior === 'function') {
```

Comment:
nit: could we use http://underscorejs.org/#isFunction ? _ is already available in the file